lib/maze/server.rb

```diff
@@ -244,6 +239,18 @@ def stop
         @thread&.kill if @thread&.alive?
         @thread = nil
       end
+
+      def reset!
+        Maze::Server.errors.clear
+        Maze::Server.sessions.clear
+        Maze::Server.builds.clear
+        Maze::Server.uploads.clear
+        Maze::Server.sourcemaps.clear
+        Maze::Server.traces.clear
+        Maze::Server.logs.clear
+        Maze::Server.invalid_requests.clear
+        @generators&.values&.each { |generator| generator.close }
```

Comment:
Can `values` ever return `nil`? If `@generators` is only `nil` or a hash, it should be fine to always call `values` here

Maybe also worth nilling out `@generators`, so there's no chance of calling a closed one

```suggestion
        @generators&.values.each(&:close)
        @generators = nil
```